src/transforms/filter.jl

```diff
@@ -69,22 +65,50 @@ DropMissing(cols::T...) where {T<:ColSelector} =
 
 isrevertible(::Type{<:DropMissing}) = true
 
-_ftrans(::DropMissing{Colon}, table) =
+# ftrans
+_ftrans(::DropMissing{Colon}, cols) =
   Filter(row -> all(!ismissing, row))
 
-function _ftrans(transform::DropMissing, table)
-  allcols = Tables.columnnames(table)
-  cols = _filter(transform.colspec, allcols)
+_ftrans(::DropMissing, cols) =
   Filter(row -> all(!ismissing, getindex.(Ref(row), cols)))
+
+# nonmissing 
+_nonmissing(::Type{T}, c) where {T} = c
+_nonmissing(::Type{Union{Missing,T}}, c) where {T} =
+  collect(T, c)
+
+function _nonmissing(columns, col)
+  c = Tables.getcolumn(columns, col)
+  _nonmissing(eltype(c), c)
 end
 
 function apply(transform::DropMissing, table)
-  ftrans = _ftrans(transform, table)
+  allcols = Tables.columnnames(table)
+  cols    = _filter(transform.colspec, allcols)
+  ftrans  = _ftrans(transform, cols)
   newtable, fcache = apply(ftrans, table)
-  newtable, (ftrans, fcache)
+
+  # post-processing
+  columns = Tables.columns(newtable)
+  newcols = [col ∈ cols ? _nonmissing(columns, col) : Tables.getcolumn(columns, col)
+             for col in allcols]
```

Comment:
Please use the do syntax to shorten up this line. The variable names in this function are a bit confusing as well. Too many cols, allcols, columns with different meanings. Can we consider other names?

Reply:
OK

Reply:
I used the Tables.columntable function to simplify the code.